### PR TITLE
PUBDEV-6056: SHAP values support for MOJO

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

**Comment by michalkurka** [comment]
@Pscheidl very early preview - please check if the APIs make sense

File: h2o-core/src/test/java/water/TestUtil.java [hunk 1]
```diff
@@ -999,4 +1000,32 @@ public void printOutColumnsMetadata(Frame fr) {
     }
   }
 
+  protected static RowData toRowData(Frame fr, String[] columns, long row) {
```
Comment (maurever) [inline h2o-core/src/test/java/water/TestUtil.java]: I am surprised why it is here right now. I have used this for loop in test many times already. :+1: ;-)

File: h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java [hunk 2]
```diff
@@ -250,7 +277,17 @@ public EasyPredictModelWrapper(Config config) {
     enableLeafAssignment = config.getEnableLeafAssignment();
     enableGLRMReconstruct = config.getEnableGLRMReconstrut();
     enableStagedProbabilities = config.getEnableStagedProbabilities();
+    enableContributions = config.getEnableContributions();
 
+    if (enableContributions) {
+      if (!(m instanceof PredictContributionsFactory)) {
+        throw new IllegalStateException("Model " + m.getClass().getName() + " cannot be used to predict contributions,");
```
Comment (maurever) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java]:
```suggestion
        throw new IllegalStateException("Model " + m.getClass().getName() + " cannot be used to predict contributions.");
```

File: h2o-genmodel/src/main/java/hex/genmodel/easy/prediction/BinomialModelPrediction.java [hunk 3]
```diff
@@ -45,5 +45,18 @@
...
    */
-  public double [] stageProbabilities;
+  public double[] stageProbabilities;
+
+  /**
+   * Per-feature prediction contributions (SHAP values).
+   * Size of the returned array is #features + 1 - there is a feature contribution column for each input feature, 
+   * the last item is the model bias. The sum of the feature contributions and the bias term is equal to the raw 
+   * prediction of the model. Raw prediction of tree-based model is the sum of the predictions of the individual
+   * trees before before the inverse link function is applied to get the actual
```
Comment (maurever) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/prediction/BinomialModelPrediction.java]:
```suggestion
   * trees before the inverse link function is applied to get the actual
```

File: h2o-genmodel/src/main/java/hex/genmodel/easy/prediction/RegressionModelPrediction.java [hunk 4]
```diff
@@ -19,4 +19,17 @@
    * Multinomial models: [probability T1.C1, probability T1.C2, ..., Tt.Cc]
    */
   public double[] stageProbabilities;
+
+  /**
+   * Per-feature prediction contributions (SHAP values).
+   * Size of the returned array is #features + 1 - there is a feature contribution column for each input feature, 
+   * the last item is the model bias. The sum of the feature contributions and the bias term is equal to the raw 
+   * prediction of the model. Raw prediction of tree-based model is the sum of the predictions of the individual
+   * trees before before the inverse link function is applied to get the actual
```
Comment (maurever) [inline h2o-genmodel/src/main/java/hex/genmodel/easy/prediction/RegressionModelPrediction.java]:
```suggestion
   * trees before the inverse link function is applied to get the actual
```

**Comment by michalkurka** [comment]
@maurever thank you!

**Pull request merged**